extensions/api/data-management/asset/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/AssetController.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022 ZF Friedrichshafen AG
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *    ZF Friedrichshafen AG - Initial API and Implementation
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.asset;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import org.eclipse.dataspaceconnector.api.datamanagement.asset.model.AssetDto;
+import org.eclipse.dataspaceconnector.api.datamanagement.asset.model.AssetEntryDto;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.query.SortOrder;
+
+import java.util.Collections;
+import java.util.List;
+
+import static java.lang.String.format;
+
+@Consumes({ MediaType.APPLICATION_JSON })
```

Comment:
Propose to use OpenAPI / swagger annotations to document more details about the endpoints, see https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations
@paullatzelsperger @jimmarino  do you agree?

Reply:
> Propose to use OpenAPI / swagger annotations to document more details about the endpoints, see https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations @paullatzelsperger @jimmarino do you agree?

Sounds like a great idea. I also like @paullatzelsperger 's suggestion to use interfaces.

Reply:
I would suggest moving this to a separate topic/PR where we can introduce these interfaces and make use of them in all existing api controllers. We should try to complete this PR asap to finally have a first draft of the openapi spec for the management part.

Reply:
there is also issue #645 where we could add this as a task